crates/router/src/db/configs.rs

```diff
@@ -177,14 +171,11 @@ impl ConfigInterface for Store {
             .await
             .map_err(|error| report!(errors::StorageError::from(error)))?;
 
-        self.get_redis_conn()
-            .map_err(Into::<errors::StorageError>::into)?
-            .publish(
-                cache::IMC_INVALIDATION_CHANNEL,
-                CacheKind::Config(key.into()),
-            )
-            .await
-            .map_err(Into::<errors::StorageError>::into)?;
+        cache::publish_into_redact_channel(
+            self.get_cache_store().as_ref(),
+            [CacheKind::Config(key.into())],
+        )
+        .await?;
```

Comment:
Same here, a stale Redis value may be left behind. Consider calling `db_call_with_redact_and_publish()` instead.